push_ios/ios/Classes/PushHostHandlers.swift

```diff
@@ -64,11 +67,11 @@ class PushHostHandlers: NSObject, PUPushHostApi {
 
     // TODO double check that the delegate is still the same later, in case the user had set it? and log error.
     init(binaryMessenger: FlutterBinaryMessenger, originalDelegate: UNUserNotificationCenterDelegate?) {
-        deviceTokenReadyDispatchGroup.enter() // DeviceToken is not yet ready
         pushFlutterApi = PUPushFlutterApi(binaryMessenger: binaryMessenger)
         delegate = UserNotificationCenterDelegateHandlers(with: originalDelegate, pushFlutterApi: pushFlutterApi)
         UNUserNotificationCenter.current().delegate = delegate
         super.init()
+        enterDeviceTokenReadyDispatchGroup() // DeviceToken is not yet ready
```

Comment:
Thanks for your PR, and I understand all your changes, except this change.

I'm curious why you moved it down from the start of the function? I can't see any issues with either case, so I'm happy to merge it.

Reply:
Hello,

If this is before `super.init` it will throw the error
```
Use of self in method call 'enterDeviceTokenReadyDispatchGroup' before super.init initialises self
```

Putting it after `super.init` fixes this.